libraries/hotstuff/chain_pacemaker.cpp

```diff
@@ -212,9 +200,22 @@ namespace eosio { namespace hotstuff {
       return n;
    }
 
+   // called from main thread
+   void chain_pacemaker::on_accepted_block( const block_state_ptr& blk ) {
+      std::scoped_lock g( _chain_state_mutex );
+      _head_block_state = blk;
+      _finalizer_set = _chain->get_finalizers(); // TODO get from chainbase or from block_state
```

Comment:
Do we have to update this on every block? Maybe check if different first?
```suggestion
      const auto& new_set = chain->get_finalizers(); // TODO get from chainbase or from block_state
      if (new_set != _finalizer_set)
         _finalizer_set = new_set;
```

Reply:
I think we will check `version` and change if `version` changes; but that is not ready yet.

Reply:
Can you add a `todo` comment?

Reply:
I did.